File: LeetCode/167.py

class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        l, r = 0, len(numbers)-1

        while l < r:
            sum = numbers[l] + numbers[r]
            
            if sum < target:
                l+=1
            elif sum > target:
                r-=1
            elif sum == target:
                return [l+1, r+1]
                    
        return[]


sol = Solution()
print(sol.twoSum([2,7,11,15], 9))

File: LeetCode/167.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l, r;
        l = 0;
        r = numbers.size();
        vector<int> ans;
        while(l<r){
            int currentSum = numbers[l]+numbers[r];
            if(currentSum < target)
                l++;
            if(currentSum > target)
                r--;
            if(currentSum == target){
                ans.push_back(l+1);
                ans.push_back(r+1);
                return ans;
            }
        }
        return ans;
    }
};